js/map.js

```diff
@@ -188,29 +217,46 @@ window.map = (function () {
     };
     isPageActive = false;
 
-    var mapPins = document.querySelectorAll('.map__pin:not(.map__pin--main)');
     mapPinsMainElement.style.top = MAP_PIN_TOP + 'px';
     mapPinsMainElement.style.left = MAP_PIN_LEFT + 'px';
     mapElement.classList.add('map--faded');
     adFormContainer.classList.add('ad-form--disabled');
+    filterTypeElement.value = 'any';
+    filterPriceElement.value = 'any';
+    filterRoomsElement.value = 'any';
+    filterGuestsElement.value = 'any';
 
-    if (mapElement.querySelector('.popup')) {
-      document.querySelector('.popup__close').removeEventListener('click', closeCardHandler);
-      document.removeEventListener('keydown', escPressCloseCardHandler);
-      removeCard();
-    }
-
-    Array.prototype.forEach.call(mapPins, function (elem) {
-      elem.parentElement.removeChild(elem);
-    });
+    window.filter.resetFilters();
+    removeCard();
+    removePins();
   }
 
+  filterTypeElement.addEventListener('change', function () {
```

Comment:
should work like that `filterTypeElement.addEventListener('change',window.debounce(applyFilter)) `

Reply:
filterTypeElement.addEventListener('change',window.debounce(applyFilter)) <- doesn't call the applyFilter function

Reply:
it's the problem with your debounce implementation. It needs to return a wrapped function. 

example: 
![image](https://user-images.githubusercontent.com/1570578/52518850-6ded1700-2c49-11e9-8b52-7d5d03f599ef.png)

Yours:
![image](https://user-images.githubusercontent.com/1570578/52518861-7f362380-2c49-11e9-9981-41f51c221599.png)

can you see the difference?

Reply:
done